pkg/rest/config_upload.go

```diff
@@ -221,63 +221,70 @@ func getExistingValuesFromEndpoint(client *http.Client, theApi api.Api, url stri
 		}
 	}
 
-	return values, nil
+	return existingValues, nil
 }
 
-func unmarshalJson(theApi api.Api, err error, resp Response, values []api.Value, objmap map[string]interface{}) (error, []api.Value) {
-
-	if theApi.GetId() == "synthetic-location" {
-
-		var jsonResp api.SyntheticLocationResponse
-		err = json.Unmarshal(resp.Body, &jsonResp)
-		if util.CheckError(err, "Cannot unmarshal API response for existing synthetic location") {
-			return err, values
-		}
-		values = translateSyntheticValues(jsonResp.Locations)
-
-	} else if theApi.GetId() == "synthetic-monitor" {
+func unmarshalJson(theApi api.Api, err error, resp Response) (error, []api.Value, map[string]interface{}) {
```

Comment:
It might make sense to move the whole `isPaginated` logic into this method,
and return some kind of `Pagnination` struct. That way we do not leak the 
`JSON` body to the outside and could add specialized pagination handling 
for different apis.

Reply:
Good input! Hmm, I'd rather see this method just doing the unmarshalling of a single response. But I could move the for loop in `getExistingValuesFromEndpoint` to its own fuction doing all of the pagination handling. Then this functionality would be encapsulated and extended if necessary in the future. Creating another abstraction for the `Pagination` is imo an overkill at this point in time. What do you think?

Reply:
I personally think it a bit strange that this method on the one hand parses `api.Values` 
and on the other hand actually return the raw json. Especially since some apis return 
JSON arrays. But I guess this is personal preference.

So I guess I'm fine with it :laughing: